geopandas/array.py

```diff
@@ -967,34 +967,54 @@ def take(self, indices, allow_fill=False, fill_value=None):
         return GeometryArray(result, crs=self.crs)
 
     def _fill(self, idx, value):
-        """Fill index locations with value
+        """
+        Fill index locations with ``value``.
 
-        Value should be a BaseGeometry
+        ``value`` should be a BaseGeometry or a GeometryArray.
         """
-        if not (_is_scalar_geometry(value) or value is None):
+        from pandas import Series
+
+        if vectorized.isna(value):
+            value = [BaseGeometry()]
+        elif _is_scalar_geometry(value):
+            value = [value]
+        elif isinstance(value, GeometryArray):
+            # If `value` is a GeoSeries, require that it has the same index as `self`.
+            if isinstance(value, Series) and not self.index.equals(value.index):
+                raise ValueError(
+                    "Index values of 'value' sequence does "
+                    "not match index values of the GeoSeries"
+                )
```

Comment:
It seems that these codes would work.
GeometryArray will transform `value` to GeometryArray.
Should I check the index in `GeoSeries.fillna`?